src/main/java/ai/deepcode/javaclient/core/AnalysisDataBase.java

```diff
@@ -577,7 +577,9 @@ private GetAnalysisResponse doGetAnalysis(
               deepCodeParams.getSessionToken(),
               bundleId,
               deepCodeParams.getMinSeverity(),
-              filesToAnalyse, HashContentUtilsBase.doGetHash(pdUtils.getFilePath(project)));
+              filesToAnalyse,
+              HashContentUtilsBase.calculateHash(project.toString())
```

Comment:
@ArtsiomCh , since `project` is an object, toString() method will return class name and hashCode (getClass().getName()+"@"+Integer.toHexString(hashCode())).
🤔 is it what we want to use for sharding or better find out the filePath?

Reply:
Correct. That's should completely fit our needs. AFAIU the use of prjDir in code-client is just for simplicity. Any string unique per project should be fine here. And so far `project.ToString()` is unique per project and stay the same within project life in Idea AFAIK.

Reply:
What if we use `project.name` here?

1. Then we don't need introduce new abstract method to obtain `project.basePath` (not exist currently)
2. Also same project located in different dirs will use the same `shard` -> same worker -> same caches. And even if some projects will have name accidentally matched - they will just share the same worker which shouldn't be the problem on backend.
@pavel-snyk WDYT?

Reply:
☝️ I like this idea. Let's do it.